src/HotChocolate/Fusion-vnext/src/Fusion.Composition/Results/CompositionResult~1.cs

```diff
@@ -0,0 +1,75 @@
+using HotChocolate.Fusion.Errors;
+
+namespace HotChocolate.Fusion.Results;
+
+public readonly record struct CompositionResult<TValue>
+{
+    public bool IsFailure { get; }
+
+    public bool IsSuccess { get; }
+
+    public List<CompositionError> Errors { get; } = [];
+
+    public TValue Value => IsSuccess
+        ? _value
+        : throw new Exception("Value may not be accessed on an unsuccessful result.");
+
+    private readonly TValue _value = default!;
+
+    private CompositionResult(TValue value)
+    {
+        _value = value;
+        IsSuccess = true;
+    }
+
+    private CompositionResult(CompositionError error)
+    {
+        Errors = [error];
+        IsFailure = true;
+    }
+
+    private CompositionResult(List<CompositionError> errors)
+    {
+        if (errors.Count == 0)
+        {
+            IsSuccess = true;
+        }
+        else
+        {
+            Errors = [.. errors];
```

Comment:
Is this syntax a conscious decision to ensure that a new list is created?

Reply:
I wrote this particular code quite a while ago, but it was probably to prevent the errors from being mutated from the outside. Using an immutable type would probably have been a better idea, which I've done now.